nucliadb_vectors/src/data_point/ops_hnsw.rs

```diff
@@ -27,41 +27,38 @@ use rand::prelude::*;
 use rand::rngs::SmallRng;
 
 use super::*;
+use crate::data_point::params;
 
-pub mod params {
-    pub fn level_factor() -> f64 {
-        1.0 / (m() as f64).ln()
-    }
-    pub const fn m_max() -> usize {
-        30
-    }
-    pub const fn m() -> usize {
-        30
-    }
-    pub const fn ef_construction() -> usize {
-        100
-    }
-}
+/// Implementors of this trait can guide the hnsw search
 pub trait DataRetriever: std::marker::Sync {
-    fn get_key(&self, _: Address) -> &[u8];
-    fn is_deleted(&self, _: Address) -> bool;
-    fn has_label(&self, _: Address, _: &[u8]) -> bool;
-    fn similarity(&self, _: Address, _: Address) -> f32;
-    fn get_vector(&self, _: Address) -> &[u8];
+    /// Returns the key of the embedding pointed by x.
+    fn get_key(&self, x: Address) -> &[u8];
+    /// Checks if x is a deleted embedding.
+    fn is_deleted(&self, x: Address) -> bool;
+    /// During filtering this function will be called to know if x has the necessary labels.
+    fn has_label(&self, x: Address, label: &[u8]) -> bool;
+    /// Computes the similarity between the embeddings pointed by x and y.
+    fn similarity(&self, x: Address, y: Address) -> f32;
+    /// Returns the serialized embedding of x, is meant to be used by [`crate::data_types::vector`].
+    fn get_vector(&self, x: Address) -> &[u8];
+    /// Embeddings with smaller similarity should not be considered.
     fn min_score(&self) -> f32;
 }
 
+/// Implementors of this trait are layers of an HNSW where a nearest neighbour search can be ran.
 pub trait Layer {
     type EdgeIt: Iterator<Item = (Address, Edge)>;
     fn get_out_edges(&self, node: Address) -> Self::EdgeIt;
 }
 
+/// Implementors of this trait are an HNSW where search can be ran.
 pub trait Hnsw {
     type L: Layer;
     fn get_entry_point(&self) -> Option<EntryPoint>;
     fn get_layer(&self, i: usize) -> Self::L;
 }
 
+/// Util for storing embedding pointers and their similarity in a [`BinaryHeap`].
```

Comment:
If `Cnx` is a data structure to store ordered embedding pointers by similarity, it's name doesn't give enough information

Reply:
`Cnx` does not store ordered embeddings, its basically a new-type for the tuple `(Address, f32)`. I had to do this because `f32` does not implement `Ord` and therefore storing the tuple directly in a `BinaryHeap` is not possible.
I fixed the documentation to express this better